source/uwp/Renderer/lib/XamlBuilder.h

```diff
@@ -11,89 +11,90 @@
-                                                     _In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderContext* renderContext,
-                                                     Microsoft::WRL::ComPtr<XamlBuilder> xamlBuilder,
-                                                     ABI::AdaptiveCards::ObjectModel::Uwp::ContainerStyle defaultContainerStyle =
-                                                         ABI::AdaptiveCards::ObjectModel::Uwp::ContainerStyle::Default) noexcept;
-        HRESULT AddListener(_In_ IXamlBuilderListener* listener) noexcept;
-        HRESULT RemoveListener(_In_ IXamlBuilderListener* listener) noexcept;
-        void SetFixedDimensions(UINT width, UINT height) noexcept;
+        void AllImagesLoaded() override;
+        void ImagesLoadingHadError() override;
+
+        static winrt::Windows::UI::Xaml::FrameworkElement
+        BuildXamlTreeFromAdaptiveCard(winrt::AdaptiveCards::ObjectModel::Uwp::AdaptiveCard const& adaptiveCard,
+                                      winrt::AdaptiveCards::Rendering::Uwp::AdaptiveRenderContext const& renderContext,
+                                      XamlBuilder* xamlBuilder,
+                                      winrt::AdaptiveCards::ObjectModel::Uwp::ContainerStyle defaultContainerStyle =
+                                          winrt::AdaptiveCards::ObjectModel::Uwp::ContainerStyle::Default);
+
+        // TODO: not sure what we need this for? Are they for public?
+        // TODO: I think these methods are created so people from outside can listen to images loading?
+        void AddListener(::AdaptiveCards::Rendering::Uwp::IXamlBuilderListener* listener);
+        void RemoveListener(::AdaptiveCards::Rendering::Uwp::IXamlBuilderListener* listener);
+        void SetFixedDimensions(uint32_t width, uint32_t height) noexcept;
         void SetEnableXamlImageHandling(bool enableXamlImageHandling) noexcept;
 
-        static HRESULT BuildPanelChildren(
-            _In_ ABI::Windows::Foundation::Collections::IVector<ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveCardElement*>* children,
-            _In_ ABI::Windows::UI::Xaml::Controls::IPanel* parentPanel,
-            _In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderContext* context,
-            _In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderArgs* renderArgs,
-            std::function<void(ABI::Windows::UI::Xaml::IUIElement* child)> childCreatedCallback) noexcept;
+        static void BuildPanelChildren(
+            winrt::Windows::Foundation::Collections::IVector<winrt::AdaptiveCards::ObjectModel::Uwp::IAdaptiveCardElement> const& children,
+            rtxaml::Controls::Panel ParentPanel,
+            rtrender::AdaptiveRenderContext context,
+            rtrender::AdaptiveRenderArgs renderArgs,
+            std::function<void(rtxaml::UIElement child)> childCreatedCallback);
```

Comment:
Can pass by const reference to avoid copy.

```suggestion
            std::function<void(rtxaml::UIElement const& child)> childCreatedCallback);
```